internal/proxy/engines/caching_policy_gen.go

```diff
@@ -0,0 +1,409 @@
+package engines
+
+// Code generated by github.com/tinylib/msgp DO NOT EDIT.
+
+import (
+	"github.com/tinylib/msgp/msgp"
+)
+
+// DecodeMsg implements msgp.Decodable
+func (z *CachingPolicy) DecodeMsg(dc *msgp.Reader) (err error) {
+	var field []byte
+	_ = field
```

Comment:
What is the purpose of this?

Reply:
This is an auto-generated code file created by `go generate` and is used internally by the msgp unmarshaler. That is true for any file ending in `_gen.go` or `_gen_test.go`

Reply:
Lol. Gotcha. It probably knows something we don't.